api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -196,4 +196,12 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
    */
   UpdateSchema deleteColumn(String name);
 
+  /**
+   * Applies all the additions, updates [renames, type widening, field documentation] and deletes
+   * from the input schema
```

Comment:
Should we call out the limitations of this API here? I don't think it is possible to support renames (since we won't know if it is delete+add instead). Deletion of a single column can probably be supported, but that combined with an addition cannot (because maybe it is a rename). So we should probably restrict ourselves to addition of field + type promotion.

Reply:
We may also say that the changes are all updates or all deletes, no mix and match. That should also be fine.

Reply:
Lets address this upstream first

Reply:
Update comment to remove renames and deletes for now?